File: src/main/java/com/mcommings/campaigner/modules/people/entities/AbilityScore.java
package com.mcommings.campaigner.modules.people.entities;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Getter
@Setter
@Builder
@Entity
@Table(name = "ability_scores")
@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
@NoArgsConstructor
@AllArgsConstructor
public class AbilityScore {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    @Column(nullable = false)
    private UUID fk_campaign_uuid;
    @Column(nullable = false)
    private int strength;
    @Column(nullable = false)
    private int dexterity;
    @Column(nullable = false)
    private int constitution;
    @Column(nullable = false)
    private int intelligence;
    @Column(nullable = false)
    private int wisdom;
    @Column(nullable = false)
    private int charisma;
}

File: src/main/frontend/lib/models/people/ability_score.dart
class AbilityScore {
  final int id;
  final String fkCampaignUuid;
  final int strength;
  final int dexterity;
  final int constitution;
  final int intelligence;
  final int wisdom;
  final int charisma;

  const AbilityScore({
    required this.id,
    required this.fkCampaignUuid,
    required this.strength,
    required this.dexterity,
    required this.constitution,
    required this.intelligence,
    required this.wisdom,
    required this.charisma,
  });

  @override
  String toString() {
    return 'AbilityScore(id: $id, fkCampaignUuid: $fkCampaignUuid, strength: $strength, dexterity: $dexterity, constitution: $constitution, intelligence: $intelligence, wisdom: $wisdom, charisma: $charisma)';
  }

  String toShortString() {
    return 'id: $id,\n strength: $strength,\n dexterity: $dexterity,\n constitution: $constitution,\n intelligence: $intelligence,\n wisdom: $wisdom,\n charisma: $charisma';
  }

  String statsOnly() {
    return 'strength: $strength,\n dexterity: $dexterity,\n constitution: $constitution,\n intelligence: $intelligence,\n wisdom: $wisdom,\n charisma: $charisma';
  }

  factory AbilityScore.fromJson(Map<String, dynamic> json) {
    try {
      return AbilityScore(
        id: json['id'],
        fkCampaignUuid: json['fk_campaign_uuid'],
        strength: json['strength'],
        dexterity: json['dexterity'],
        constitution: json['constitution'],
        intelligence: json['intelligence'],
        wisdom: json['wisdom'],
        charisma: json['charisma'],
      );
    } catch (e) {
      throw FormatException('AbilityScore parse error: $e');
    }
  }
}
